### DISCUSSION TITLE
Is it possible to get cuda or cuda containers running without cuda-samples?

### DISCUSSION THREAD

**Discussion by pdelbarba** [OE4T/meta-tegra#1192] [Q&A]
Currently I have:
```
    cuda-driver \
    cuda-cudart \
    nvidia-docker \
```

`docker run --rm --runtime nvidia <image>`
results in 
`docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #1: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: driver error: failed to process request: unknown.`

*however*, including `cuda-samples` causes it to run fine but appears to add ~500MB to the rootfs size and I'm very disk space sensitive.

Is there a particular document I should be looking at for this?  If not, let me know and I'll gladly write something up once I get this working.

UPDATE: I was able to get this working by adding `cuda-libraries` to my IMAGE_INSTALL in addition to the above.  I'm not sure if this is the correct solution and it seems odd that this isn't already a dependency of cuda-driver or cuda-cudart but at least it works.  Is this the intended way to include these?

**Comment by madisongh** [comment]
> UPDATE: I was able to get this working by adding cuda-libraries to my IMAGE_INSTALL in addition to the above. I'm not sure if this is the correct solution and it seems odd that this isn't already a dependency of cuda-driver or cuda-cudart but at least it works. Is this the intended way to include these?

Yes.  `cuda-libraries` is an empty package that RDEPENDS on all of the CUDA library packages.  Those, in turn depend on the CUDA runtime and driver.
